src/Core/Migration/V6_6/Migration1701776095FixDefaultMailFooter.php

```diff
@@ -0,0 +1,83 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Migration\V6_6;
+
+use Doctrine\DBAL\ArrayParameterType;
+use Doctrine\DBAL\Connection;
+use Shopware\Core\Content\MailTemplate\Aggregate\MailHeaderFooterTranslation\MailHeaderFooterTranslationDefinition;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\Framework\Migration\MigrationStep;
+use Shopware\Core\Framework\Uuid\Uuid;
+use Shopware\Core\Migration\Traits\ImportTranslationsTrait;
+
+/**
+ * @internal
+ */
+#[Package('core')]
+class Migration1701776095FixDefaultMailFooter extends MigrationStep
+{
+    use ImportTranslationsTrait;
+
+    public function getCreationTimestamp(): int
+    {
+        return 1701776095;
+    }
+
+    public function update(Connection $connection): void
+    {
+        $plainDe = (string) file_get_contents(__DIR__ . '/../Fixtures/mails/defaultMailFooter/de-plain.twig');
+
+        $languages = $this->getLanguageIds($connection, 'de-DE');
+        if (!$languages) {
+            return;
+        }
+
+        $translations = $this->getTranslationIds($connection, $languages);
+        if (empty($translations)) {
+            return;
+        }
+
+        foreach ($translations as $translation) {
+            $connection->update(
+                MailHeaderFooterTranslationDefinition::ENTITY_NAME,
+                [
+                    'footer_plain' => $plainDe,
+                ],
+                [
+                    'language_id' => $translation['language_id'],
+                    'mail_header_footer_id' => $translation['mail_header_footer_id'],
+                ]
+            );
+        }
+    }
+
+    public function updateDestructive(Connection $connection): void
+    {
+        // implement update destructive
+    }
+
+    /**
+     * @param array<string> $languageIds
+     *
+     * @return mixed[]
+     */
+    private function getTranslationIds(Connection $connection, array $languageIds): array
+    {
+        return $connection->fetchAllAssociative(
+            '
+            SELECT mail_header_footer_translation.mail_header_footer_id, mail_header_footer_translation.language_id
+
+            FROM mail_header_footer
+
+                INNER JOIN mail_header_footer_translation
+                    ON mail_header_footer.id = mail_header_footer_translation.mail_header_footer_id
+
+            WHERE mail_header_footer_translation.language_id IN (:ids)
+            AND mail_header_footer.system_default = 1
```

Comment:
Hey @keulinho 
I'm working on the migration and used the mentioned PR as inspiration.
Can't use it 1:1 as this is no normal mail template, but a mail_header_footer template.

To detect the correct template I assumed checking for the system_default flag should do the trick: `mail_header_footer.system_default = 1`
**But** that'S not the case. actually it is set to zero for the default mail header footer on a fresh installation.

So, should I use a 'LIKE %Addresse:%' instead?

Thanks for your advice!

Reply:
Can you add a migration first that hashes the content of the template with md5 and compares that to the "old" template content and then set the `system_default` accordingly to `true`

And also adjust the original migration that adds the template to also set the system default to true, then you can rely on that here